src/lib.rs

```diff
@@ -105,7 +105,7 @@ impl AttributeDataType {
         Ok(attr)
     }
 
-    pub fn size_of(&self) -> usize {
+    pub fn size_of(self) -> usize {
```

Comment:
Why consume `self`? We just want to know its size...

Reply:
Ah, that's because above `Copy` was introduced. What do we need `Copy` for?

Reply:
I don't think it was needed per se, but for small structs, it's a small convenience and efficiency win with no downsides.

Reply:
Copy was added for the attribute+xyz iterator returned from S2Meta